vcsutils/webhookparser/factory.go

```diff
@@ -6,16 +6,40 @@ import (
 	"github.com/jfrog/froggit-go/vcsutils"
 )
 
-func createWebhookParser(provider vcsutils.VcsProvider, request *http.Request) WebhookParser {
-	switch provider {
+// ParserBuilder builds VcsClient
+type ParserBuilder struct {
+	vcsProvider vcsutils.VcsProvider
+	request     *http.Request
+	endpoint    string
+}
+
+// NewParserBuilder creates new ParserBuilder
+func NewParserBuilder(vcsProvider vcsutils.VcsProvider, request *http.Request) *ParserBuilder {
```

Comment:
It looks like this function can be made provate. Let's check the other functions and type in this file.

Reply:
Actually, it must be public.
We need it so that we can provide an API endpoint (used for generating compare URLs).

Reply:
NP.

Reply:
BTW, we may consider moving the Parse functio to the VCS client type, since it may use the API endpoint.
WDYT?

Reply:
@ cyrilc-pro - before answering your above questions, I look at the code again, and still don't understand why `NewParserBuilder` should be public. It used only from within its own package.

Reply:
We use it in our Integration Service. Please DM me if you want me to show you.

Reply:
@cyrilc-pro - wouldn't be better if yuo use the API that is documented in the README? This way, it'll never be broken, and we'll be able to add make the above API private. Froggit-go is supposed to expose it's public APIs through the README only. There's no other "memory" for APIs that are used by other applications and should not be broken.

Reply:
But I _need_ to be able to provide the API endpoint when parsing the webhook event. The existing function `ParseIncomingWebhook` does not allow this without breaking the API. Hence the solution I came up with.

Reply:
If you prefer, I can remove this code, and I'll set the URL in our Integration Service. Thus, I won't touch the existing API.

Reply:
I think that you stepped on a limitation in the existing pubic API (`ParseIncomingWebhook`). Isn't it the case?
We should fix this for the benefit of all users of `froggit-go` and not just the Integration Service that uses it. If this fix/improvement requires breaking the API, then. we should do it. The release notes will highlight this change as a breaking change.

Reply:
According to our discussion, I introduced breaking changes to the `ParseIncomingWebhook` function. The new signature is described in the README.
Please note that I also introduced "provisional" args: `ctx` and `logger`. They are not in use yet, but they might become handy at some point.